### PR TITLE
Using Microsoft.IO.RecyclableMemoryStream instead of "new MemoryStream" to reduce LOH and GC

### LINKED ISSUE #82: Utilize recyclable memory stream instead of "new MemoryStream" all over
It would be nice if you guys utilized a recyclable memory stream to reduce LOH and GC

https://github.com/Microsoft/Microsoft.IO.RecyclableMemoryStream

There is a package here that makes this super easy.

### PR DESCRIPTION
**Pull request by LIFEfreedom**
I tried to replace the use of "new MemoryStream" in all places, but couldn't come up with anything with that in the OfficeOpenXml.Packaging namespace (ZipPackage.cs and ZipPackagePart.cs) and in the ExcelWorksheet.cs file (CreateXml () function)
Fixes #82

### CODE REVIEW

**Comment by JanKallman** [comment]
Hello, before I start to look at this I need you to sign our [CAA](https://epplussoftware.com/legal/caa) as this pull request exceeds 20 lines of signigicant code. 
If you are ok with that, please sign it and mail it to us.
For more info see our https://github.com/EPPlusSoftware/EPPlus/blob/develop/CONTRIBUTING.md .

**Comment by LIFEfreedom** [comment]
@JanKallman I signed the agreement and sent it to the mail

**Comment by JanKallman** [comment]
@LIFEfreedom Thanks, I'll try to verify it in the coming days.

File: src/EPPlus/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -46,4 +46,8 @@
     //[assembly: AssemblyVersion("5.2.1.0")]
     //[assembly: AssemblyFileVersion("5.2.1.0")]
 #endif
-[assembly: AllowPartiallyTrustedCallers]
\ No newline at end of file
+[assembly: AllowPartiallyTrustedCallers]
+
+#if !NET35
+[assembly: SecurityRules(SecurityRuleSet.Level1)]
```
Comment (JanKallman) [inline src/EPPlus/Properties/AssemblyInfo.cs:52]: Why this change?

**Review by JanKallman — CHANGES_REQUESTED** [review]

File: src/EPPlus/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (LIFEfreedom) [reply src/EPPlus/Properties/AssemblyInfo.cs:52]: the security policy in the .Net framework did not allow some code to execute, I was getting this error Method `"OfficeOpenXml.Utils.RecyclableMemory.GetStream ()" could not access method "OfficeOpenXml.Utils.RecyclableMemory.get_MemoryManager ()".` at https://github.com/EPPlusSoftware/EPPlus/pull/207/files?file-filters%5B%5D=.cs#diff-88ed0abfe303a02040995b05494c6bf79414df00ecb89f678f8eab16d7d6f092R33
As far as i understood it is due to missing `[assembly: AllowPartiallyTrustedCallers]` in **Microsoft.IO.RecyclableMemoryStream** library

Comment (JanKallman) [reply src/EPPlus/Properties/AssemblyInfo.cs:52]: Yes, I see, but this setting revert the security back to .NET 2.0 and should not be used. 
https://docs.microsoft.com/en-us/dotnet/framework/misc/security-transparent-code-level-1
I have opened an issue to see if this can be changed... https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream/issues/108

Comment (LIFEfreedom) [reply src/EPPlus/Properties/AssemblyInfo.cs:52]: Subscribed to notifications, I will follow the issue

Comment (LIFEfreedom) [reply src/EPPlus/Properties/AssemblyInfo.cs:52]: Immediately after the adoption of the [PR](https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream/pull/110), I will remove this lines

**Review by JanKallman — APPROVED** [review]

**Comment by JanKallman** [comment]
Great, I will have another look at this and merge it for version 5.5, with estimated release in a few weeks. Nice work!

**Comment by JanKallman** [comment]
Merged, Thanks a lot!
